www_src/pages/map/map.jsx

```diff
@@ -1,66 +1,170 @@
+
+    this.cartesian = new Cartesian({
+      allCoords: elements.map(el => el.coords),
+      width,
+      height,
+      gutter
+    });
+  },
+  componentDidMount: function () {
+    var el = this.getDOMNode();
+    var bounding = this.refs.bounding;
+    var boundingEl = bounding.getDOMNode();
+    var startX, startY, deltaX, deltaY;
+    var didMove = false;
+
+    // Set up initial translation
+    this.setState({
+      camera: this.cartesian.getFocusTransform({x: 0, y: 0})
+    });
+
+    el.addEventListener('touchstart', (event) => {
+      didMove = false;
+      startX = event.touches[0].clientX;
+      startY = event.touches[0].clientY;
+      boundingEl.style.transition = 'none';
+    });
+
+    el.addEventListener('touchmove', (event) => {
+      didMove = true;
+
+      var x = parseInt(this.state.camera.x);
+      var y = parseInt(this.state.camera.y);
+
+      deltaX = (event.touches[0].clientX - startX) / this.state.zoom;
+      deltaY = (event.touches[0].clientY - startY) / this.state.zoom;
+
+      var translation = 'translate(' + (x + deltaX) + 'px, ' + (y + deltaY) + 'px)';
+      boundingEl.style.transform = translation;
+
+    });
+
+    el.addEventListener('touchend', (event) => {
+      boundingEl.style.transition = '';
+      if (didMove) {
+        this.state.camera.x += deltaX;
+        this.state.camera.y += deltaY;
+      }
+    });
+  },
+  selectPage: function (el) {
+    return () => {
+      var state = {
+        camera: this.cartesian.getFocusTransform(el.coords),
+        selectedEl: el.id,
+      };
+      if (this.state.selectedEl === el.id) {
+        state.zoom = 1;
+      }
+      this.setState(state);
     };
   },
-  contextTypes: {
-    router: React.PropTypes.func
+  zoomOut: function () {
+    this.setState({zoom: this.state.zoom / 2});
   },
-  zoomGridOut: function () {
-    this.refs.masterGrid.zoomOut();
+  zoomIn: function () {
+    this.setState({zoom: this.state.zoom * 2});
   },
-  zoomGridIn: function () {
-    this.refs.masterGrid.zoomIn();
+  edit: function () {
+    // todo - go to editor
   },
-  componentDidMount: function () {
-    // Pass container dimensions in once initial render is complete
-    //   since container must be measured before tiles can be properly laid out...
-    var elWrapper = this.refs.wrapper.getDOMNode();
-    this.refs.masterGrid.setContainerDimensions(elWrapper.clientWidth, elWrapper.clientHeight);
-
-    setTimeout(function() {
-      this.refs.masterGrid.setZoomLevel(this.state.zoomLevel);
-    }.bind(this), 100);
+  addPage: function (coords) {
+    return () => {
+      var id = 'foo' + elements.length
+      elements.push({id, coords: coords, text: 'wahoo...'});
+      this.cartesian.allCoords.push(coords);
+      this.setState({
+        camera: this.cartesian.getFocusTransform(coords),
+        selectedEl: id
+      });
+    };
   },
-  onZoomChange: function (event) {
+  removePage: function () {
+    var index;
+    elements.forEach((el, i) => {
```

Comment:
This is kinda why I feel a little weird about this new schema. I almost wonder if we should have our keys just be IDs in an object instead of using an array. Having to iterate over the array to lookup by ID seems not especially performant. I don't think order matters either?

Like:

```javascript
{
  'guid12345': {
    coords: {x: 0, y: 0},
    text: 'Hello'
  },
  'guid52353': {
    coords: {x: 1, y: 0},
    text: 'Bye'
  }
}
```